### PR TITLE
[ENHANCEMENT] Add options to change how winner name is shown

### PR DESCRIPTION
**Pull request by mickeystix**
Addition of a new config.lua with options to change how a race winners name is presented upon victory.

Options are; No Name, Char Name, Player Name.
If no selections are made or are all false in config, it will use the player/account name, which is the default behavior,

Account/Player Name:
image

Character/Ped Name:
image

No Name: 
image

Also made some consistency changes where different currencies were being shown, and some grammar fixes.

**Questions (please complete the following information):**
- Have you personally loaded this code into an updated qbcore project and checked all it's functionality? [yes] (Be honest)
- Does your code fit the style guidelines? [yes]
- Does your PR fit the contribution guidelines? [yes]

### CODE REVIEW

File: config.lua [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Config = Config or {}
+
+Config.UseNoName = true -- Dont use any name when someone wins
+Config.UseCharName = false -- Use the character name when announcement is made on win
```
Comment (Z3rio) [inline config.lua:4]:
```suggestion
Config.UseCharName = false -- Uses the character name if true, if false it uses the steam / default name.
```
The explenation for this line is quite incorrect, this explenation would fit it alot better.

File: server/main.lua [hunk 2]
```diff
@@ -36,13 +42,14 @@ RegisterNetEvent('qb-streetraces:JoinRace', function(RaceId)
...
                 TriggerClientEvent('qb-streetraces:SetRaceId', src, RaceId)
+                TriggerClientEvent('qb-streetraces:BeginPhasing', src)
                 TriggerClientEvent('QBCore:Notify', zPlayer.PlayerData.source, GetPlayerName(src).." Joined the race", 'primary')
             end
         else
             TriggerClientEvent('QBCore:Notify', src, "You dont have enough cash", 'error')
         end
     else
-        TriggerClientEvent('QBCore:Notify', src, "The person wo made the race is offline!", 'error')
+        TriggerClientEvent('QBCore:Notify', src, "The person who made the race is offline!", 'error')
```
Comment (Z3rio) [inline server/main.lua:52]:
```suggestion
        TriggerClientEvent('QBCore:Notify', src, "The person which made the race is offline!", 'error')
```
Sounds better

**Review by Z3rio — CHANGES_REQUESTED** [review]
The code should work as it is right now, although I would still like to request these changes since they would make this PR even better.

File: server/main.lua [hunk 3]
```diff
@@ -102,7 +108,7 @@ function CancelRace(source)
                     for _, iden in pairs(Races[key].joined) do
                         local xdPlayer = QBCore.Functions.GetPlayer(iden)
                             xdPlayer.Functions.AddMoney('cash', Races[key].amount, "race-cancelled")
-                            TriggerClientEvent('QBCore:Notify', xdPlayer.PlayerData.source, "Race Has Stopped, You Got Back $"..Races[key].amount.."", 'error')
+                            TriggerClientEvent('QBCore:Notify', xdPlayer.PlayerData.source, "Race Has Stopped, You Got Back $"..Races[key].amount..".", 'error')
```
[the remark this answers is not in the document]
Comment (mickeystix) [reply server/main.lua:111]: Sorry I didn't see your comments! All great! I can make these changes.

**Review by GodyFromDiscord — CHANGES_REQUESTED** [review]
Please update requested changes from Zerio & Myself

**Comment by ChristianBDev** [comment]
@mickeystix Please fix and update the requested changes so we may be able to push up.

LGTM

File: server/main.lua [hunk 4]
```diff
@@ -12,17 +11,24 @@ RegisterNetEvent('qb-streetraces:NewRace', function(RaceTable)
...
+    local name =  (xPlayer.PlayerData.charinfo.firstname .. " " .. xPlayer.PlayerData.charinfo.lastname)-- Get Character name
     xPlayer.Functions.AddMoney('cash', Races[RaceId].pot, "race-won")
-    TriggerClientEvent('QBCore:Notify', src, "You won the race and €"..Races[RaceId].pot..",- recieved", 'success')
+    TriggerClientEvent('QBCore:Notify', src, "You won the race and $"..Races[RaceId].pot.."!", 'success')
     TriggerClientEvent('qb-streetraces:SetRace', -1, Races)
-    TriggerClientEvent('qb-streetraces:RaceDone', -1, RaceId, GetPlayerName(src))
+    if Config.UseCharName then
+        TriggerClientEvent('qb-streetraces:RaceDone', -1, RaceId, name) -- Use Character name
+    elseif Config.UsePlayerName then
+        TriggerClientEvent('qb-streetraces:RaceDone', -1, RaceId, GetPlayerName(src)) -- Use Account name
```
[the remark this answers is not in the document]
Comment (mickeystix) [reply server/main.lua:28]: This is intentional to allow for player name as default even if someone goofs up the config and enters incorrect/unexpected values.

[the remark this answers is not in the document]
Comment (mickeystix) [reply server/main.lua]: Removed redundancy

File: config.lua [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+Config.UsePlayerName = false -- Whatever identifier you use aka steamid, fivem, etc
```
[the remark this answers is not in the document]
Comment (mickeystix) [reply config.lua]: Removed dead bool now that this is unused

File: fxmanifest.lua [hunk 6]
```diff
@@ -6,5 +6,6 @@ version '1.2.0'
 
 client_script 'client/main.lua'
 server_script'server/main.lua'
+shared_scripts {'config.lua'}
```
[the remark this answers is not in the document]
Comment (mickeystix) [reply fxmanifest.lua]: Removed redundancy/safety net checks and dead booleans from config. 
Swapped manifest for individual script reference.

File: fxmanifest.lua [hunk 7]
```diff
@@ -6,5 +6,6 @@ version '1.2.0'
 
 client_script 'client/main.lua'
 server_script'server/main.lua'
```
Comment (ChristianBDev) [inline fxmanifest.lua:8]: Change to `server_script 'server/main.lua'`

File: server/main.lua [hunk 8]
```diff
@@ -36,13 +40,14 @@ RegisterNetEvent('qb-streetraces:JoinRace', function(RaceId)
             if xPlayer.Functions.RemoveMoney('cash', Races[RaceId].amount, "streetrace-joined") then
                 TriggerClientEvent('qb-streetraces:SetRace', -1, Races)
                 TriggerClientEvent('qb-streetraces:SetRaceId', src, RaceId)
+                TriggerClientEvent('qb-streetraces:BeginPhasing', src)
```
Comment (ChristianBDev) [inline server/main.lua:43]: Not sure what this points too, need to remove.

**Review by ChristianBDev — CHANGES_REQUESTED** [review]
Review and reasses changes made, Not sure what BeginPhasing is as I have searched it up but I dont see it anywhere in code. Please explain

File: server/main.lua (hunk 8 above)
Comment (mickeystix) [reply server/main.lua]: Phasing in street racing in GTARP is removing collision between racers. I use it in custom version. This is a remnant. Removed.

**Pull request closed without merging**